convert_gvf_to_vcf/vcfline.py

```diff
@@ -271,5 +260,181 @@ def get_alt(self, field_lines_dictionary, all_possible_lines_dictionary):
+                                                                                                '.'))  # adds missing values if not found
+                sample_format_value_tokens.append(":".join(format_value_list))
+            else:
+                sample_format_value_tokens.append(':'.join(['.' for key in list_of_format_key] or ['.']))
+        self.format_values_by_sample_string = '\t'.join(sample_format_value_tokens)
+        return self.format_values_by_sample_string
+    # functions responsible for INFO are below
+    def convert_info_list_to_dict(self):
+        """ This converts list of INFO fields in a VCF line to a dictionary. This will be useful when merging fields of a VCF line.
+        :return: info_dict: converted dictionary of INFO fileds
+        """
+        info_dict = {}
+        for i in self.info:
+            tokens = i.split(";")
+            for token in tokens:
+                info_key, info_value = token.split("=")
+                info_dict[info_key] = info_value
+        # NOTE: info_dict is a contains same info as vcf_value except the keys are converted
+        return info_dict
+
+    def merge_info_dicts(self, other_vcf_line):
+        """ Merges and stores the INFO dictionaries for the INFO field of a VCF line.
+        :param: other_vcf_line
+        """
+        merged_info_dict = {}
+        for key in self.info_dict.keys() | other_vcf_line.info_dict.keys():
+            self.value_info_dict = self.info_dict.get(key)
+            other_vcf_line.value_info_dict = other_vcf_line.info_dict.get(key)
+            if self.value_info_dict == other_vcf_line.value_info_dict:
+                merged_info_dict[key] = self.value_info_dict
+            else:
+                if self.value_info_dict is None:
+                    merged_info_dict[key] = other_vcf_line.value_info_dict
+                elif other_vcf_line.value_info_dict is None:
+                    merged_info_dict[key] = self.value_info_dict
+                else:
+                    merged_info_dict[key] = f"{self.value_info_dict},{other_vcf_line.value_info_dict}"
+        # Store merged info dict for this VCF line and the other VCF line.
+        self.info_dict = merged_info_dict
+        other_vcf_line.info_dict = merged_info_dict
+
+    def format_info_string(self):
+        """ Creates a formatted INFO string using the INFO dictionary. Anchors ID to start of the string.
+        :return: info_string: formatted INFO string for use in VCF line
+        """
+        info_parts = []
+        if "ID" in self.info_dict:
+            info_parts.append(f"ID={self.info_dict["ID"]}")
+        for key, value in self.info_dict.items():
+            if key != "ID":
+                info_parts.append(f"{key}={value}")
+        self.info_string = ";".join(info_parts)
+        return self.info_string
+
+    # MERGE OR KEEP below
+    def merge(self, other_vcf_line, list_of_sample_names):
```

Comment:
The `merge` and `keep` functions are returning a set of strings ready to be printed where I think you should keep the information in the VcfLine  object until the end and print them using the `__str__`